codegen/templates/http_client.tmpl

```diff
@@ -206,7 +229,24 @@ func (c *{{$clientName}}) {{$methodName}}(
 
 	{{if $sidecarRouter -}}
 	headers[c.callerHeader] = c.callerName
-	headers[c.calleeHeader] = c.calleeName
+
+	// Set the service name if dynamic routing header is present
```

Comment:
I realized I should invert this... iterate through alternative routes and see if header is present in header map


There will be less routes than headers generally (most often 0!)

Reply:
yes. good idea. 👍